Source/WebCore/platform/mock/mediasource/MockMediaPlayerMediaSource.cpp

```diff
@@ -205,6 +206,11 @@ MediaTime MockMediaPlayerMediaSource::currentMediaTime() const
     return m_currentTime;
 }
 
+bool MockMediaPlayerMediaSource::currentMediaTimeMayProgress() const
+{
+    return MediaSourcePrivate::hasFutureTime(currentMediaTime(), durationMediaTime(), *buffered());
```

Comment:
Ditto, can we move this method to the base class?

Reply:
There's already a default implementation, which is based on the readyState.